### PR TITLE
kata-deploy: for testing, make sure we use the PR branch

### PR DESCRIPTION
**Pull request by egernst**
## Changes

This pull request makes two changes to the kata-deploy worfklows.

1. For kata-deploy-test, let's make sure we are exercising changes that are introduced by the PR itself, rather than our base. With this, you can expect changes to our build scripts and/or kata-deploy scripts to be exercised. This doesn't change behavior, however, if changes to the kata-deploy-test workflow itself. Developers will need to continue using a fork to verify these (which is how this PR is tested - details below).
2. For kata-deploy-push:
 *  let's make sure that we only run the workflow if the PR includes changes to ./tools/**. This way, we don't waste cycles for every single PR, and only exercise for use case that we care about (ensuring that ./tools/packaging and ./tools/osbuilder are still running correctly).  This'll reduce our load on GitHub Actions, and should reduce our average queue time for jobs.

## Testing

To verify the change in behavior for (1): 
- In this repo, verify existing behavior, showing that we don't exercise changes, as described in (1). See https://github.com/kata-containers/kata-containers/actions/runs/1781160146, which shows the pipeline succeeding, despite my test PR which breaks the build. 
- In my fork, which is enabled to run the tests, I set *my* main to match this PR. Running same PR on top of this, we can see that the builds are failing, and that the GitHub action is catching this failure: See https://github.com/egernst/kata-containers/runs/5029448370?check_suite_focus=true

### ** Additional test **
To ensure this _also_ works on non-main branch, I tested on my fork, instead with a PR to stable-foobar:
 - action run: https://github.com/egernst/kata-containers/runs/5039331801?check_suite_focus=true#step:5:13
 - PR: https://github.com/egernst/kata-containers/pull/14

To verify (2):
- I have shown following PR which will just run on main without checks, for non ./tools paths:  https://github.com/kata-containers/kata-containers/pull/3588
- With PR in place, we can see that workflow changes are not exercised if path doesn't include ./tools/** in [1] , and that the job will run if tools is touched in [2].
[1] - https://github.com/egernst/kata-containers/pull/12
[2] - https://github.com/egernst/kata-containers/pull/13

Signed-off-by: Eric Ernst <eric_ernst@apple.com>

### CODE REVIEW

**Comment by fidencio** [comment]
> /test_kata_deploy

This won't have any effect as the action tested is the one that's part of the repo.
In order to test this change, I'd like to ask you to:
* push this to your `main`.
* let someone else open a PR against your main, that has this yaml file committed.

For now, let me add a `do-not-merge` label to ensure this gets properly tested.

File: .github/workflows/kata-deploy-push.yaml [hunk 1]
```diff
@@ -9,7 +9,8 @@ on:
       - synchronize
       - labeled
       - unlabeled
-  push:
+    paths:
+      - tools/**
```
Comment (fidencio) [inline .github/workflows/kata-deploy-push.yaml:13]: @egernst, I'd prefer having this is a different PR, so we can have more discussions on whether or not to keep this action.

**Comment by snir911** [comment]
/test-fedora
It seems to me it's working as expected, commenting on a PR on a non-main branch made the workflow to pull the actual PR.
However!, i was not able to test it properly so i tested a degraded version of the workflow, if someone will be able to run it through the complete workflow it would be better

**Comment by fidencio** [comment]
@snir911, where you able to test (even with the dummy workflow) on a branch different than the `main` one?

**Comment by snir911** [comment]
Yes, I was using the PR link git presents when i pushed a temporary branch to my own repo and just set it to be a PR on the non-main branch

**Comment by egernst** [comment]
I have tests linked above as well @snir911 - you should be able to see the test against master. As well as duplicated pipeline in my org. 

@fidencio I don’t think the tools path filter is too controversial. If we also want it removed, we can follow up?

**Comment by fidencio** [comment]
@snir911, @egernst, can you please paste the link of a run of `/test_kata_deploy` running against a branch that's different than the `main` one?  Similarly to what's been done the kata-deploy-push tests.

That would simplify immensely the life of other reviewers.

**Comment by egernst** [comment]
@fidencio done - it behaves same as main, since its just checking out the PR.  Updated in PR description under testing. Link here, too: https://github.com/egernst/kata-containers/runs/5039331801?check_suite_focus=true#step:5:13

**Review by fidencio — APPROVED** [review]
lgtm, thanks @egernst!

**Review by snir911 — DISMISSED** [review]
Thanks!

File: .github/workflows/kata-deploy-push.yaml (hunk 1 above)
Comment (wainersm) [reply .github/workflows/kata-deploy-push.yaml:13]: Does kata-deploy read values from `versions.yaml`? If yes, then I think that file should be included in the list of paths.

Comment (egernst) [reply .github/workflows/kata-deploy-push.yaml:13]: The build scripts do reference -- good suggestion.

File: .github/workflows/kata-deploy-test.yaml [hunk 2]
```diff
@@ -48,18 +48,16 @@ jobs:
           - rootfs-initrd
           - shim-v2
     steps:
-      # As Github action event `issue_comment` does not provide the right ref
```
Comment (wainersm) [inline .github/workflows/kata-deploy-test.yaml:51]: I suggest to rephrase that comment and keep it as it helps a lot to understand why the heck it needs to get the ref.

**Review by wainersm — APPROVED** [review]
Hi @egernst ,
I don't have any further comments. Thanks for addressing these issues!

**Review by fidencio — APPROVED** [review]
lgtm, thanks @egernst!

**Pull request merged**